api/dataprocessor.go

```diff
@@ -188,6 +188,8 @@ func (s *Server) getTargets(reqs []models.Req) ([]models.Series, error) {
 	if len(localReqs) > 0 {
 		wg.Add(1)
 		go func() {
+			// the only errors returned are from us catching panics, so we should treat them
+			// all as internalServerErrors
```

Comment:
1) dataprocessor.go is used exclusively by the http API methods, so it belongs in the api package.
2) These should be probably be standardized.  But to me "Dataprocessing" doesnt mean anything as all of these functions are used exclusively for handling the graphite /render calls.
3) this doesnt make any sense to me and just looks like you are trying to create problems that dont exist.  a) of course a cli tool can return differnt types of erors. b) the cli tool will have its own features and code base and so can handle errors how ever it wants.